### ISSUE TITLE
Remove image for key is not working

### ISSUE THREAD

**Issue by karnakar** [rs/SDWebImage#1636]
Hi,

I'm using refreshCache options in sdimage download to refresh the image but this does not loading new image from url. I also tried remove image  for key for removing image from cache by using url as key. I tried two ways but it still shows old image i.e, from cache but not loading new image 

### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           |  ios
 Platform Version        | 9.0
 SDWebImage Version      |  3.7.6
 Integration Method      | cocoapods
 Xcode Version           |  Xcode 7.3
 Repro rate              |  all the time (100%) / sometimes x% / only once

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by bpoplauschi** [comment]
Please provide some example image urls. Also, notice the latest version out there is 3.8.2, might want to try upgrading first.
